### PR TITLE
add Foreman Status entity

### PR DESCRIPTION
**Pull request by sean797**
```
>>> config = ServerConfig(url="https://centos7-katello-nightly.sean.example.com", auth=( 'admin', 'changeme'), verify=False)
>>> ForemanStatus(config).search_json()
{u'status': 200, u'version': u'1.20.0-develop', u'result': u'ok', u'api_version': 2}

### CODE REVIEW

**Comment by akofink** [comment]
I would love to rename the old Status to KatelloStatus, but I guess that would break stuff, eh?

**Comment by sean797** [comment]
So I've taken a slightly different tactic which also solves the issues outlined in https://github.com/SatelliteQE/nailgun/pull/515. If people are okay with the approach we can make changes to `Ping` & maybe other classes in another PR. ping @abalakh

@akofink I've added `KatelloStatus` I'm all for just dropping `Status` without `EntityReadMixin` or  `EntitySearchMixin` I don't think there is much of a use-case for it. Opinions, please?

**Comment by akofink** [comment]
@SatelliteQE please make sure this change won't break anything.

@sean797 I still think, even if QE says this doesn't break stuff, we shouldn't change nailgun's API. Let's assume people are using the Status class for Satellite, which includes Katello. We shouldn't expect them to change their code because of this update to nailgun. I'd still prefer some form of alias.

**Comment by sean797** [comment]
ping?

**Comment by jyejare** [comment]
@sean797 , Looks good to me but needs to test this on Satellite 6.5.  Also, the result in https://github.com/SatelliteQE/nailgun/pull/529#issue-209662923 seems to be taken with SearchMixin and not ReadMixin as the PR has been updated (I think so) over the period. 

Please update the results from 6.5 nightly if you do have 6.5 handy.

**Comment by sean797** [comment]
Thanks @jyejare, no, I don't have Satellite, my changes are generally driven by https://github.com/theforeman/foreman-ansible-modules and therefore I test with Foreman or Katello.

**Comment by jyejare** [comment]
> @sean797 , Looks good to me but needs to test this on Satellite 6.5. Also, the result in [#529 (comment)](https://github.com/SatelliteQE/nailgun/pull/529#issue-209662923) seems to be taken with SearchMixin and not ReadMixin as the PR has been updated (I think so) over the period.
> 
> Please update the results from 6.5 nightly if you do have 6.5 handy.

@sean797 , sorry for the confusion of 6.5. Actually I am asking you to post results from 6.5 upstream nightly or your devel environment with read_json(). 

Currently, The upstream nightly is broken.

File: nailgun/entity_mixins.py [hunk 1]
```diff
@@ -388,9 +388,10 @@ def __init__(self, server_config=None, **kwargs):
         # `super`, but that's not always the case.
         if not hasattr(self, '_fields'):
             self._fields = {}
-        self._fields.setdefault('id', IntegerField(unique=True))
         if not hasattr(self, '_meta'):
             self._meta = {}
+        if 'read_type' not in self._meta or self._meta['read_type'] != 'base':
```
Comment (jyejare) [inline nailgun/entity_mixins.py]: Here:
```suggestion
        if 'read_type' not in self._meta':
```
is enough.

File: nailgun/entity_mixins.py [hunk 2]
```diff
@@ -723,8 +724,13 @@ def read_raw(self, params=None):
         :return: A ``requests.response`` object.
 
         """
+        if 'read_type' in self._meta:
```
Comment (jyejare) [inline nailgun/entity_mixins.py]:
```suggestion
        'base' if 'read_type' in self._meta else 'self'
```

**Comment by sean797** [comment]
Thanks @jyejare I've wrote it those aways to allow for different `read_type`'s in the future. Happy to update this with your suggestions if required?

**Comment by jyejare** [comment]
@sean797 , Yeah It wasnt a blocker comment but its good to keep the code as small as possible. :)

File: nailgun/entity_mixins.py (hunk 2 above)
Comment (ldjebran) [inline nailgun/entity_mixins.py]: perhaps: 
```python
path_type = self._meta.get('read_type', 'self')

```

**Comment by akofink** [comment]
Note these PRs in progress:

https://github.com/theforeman/foreman/pull/6134
https://github.com/Katello/katello/pull/7751

**Comment by sean797** [comment]
That works, thanks @ldjebran :+1:

**Comment by ldjebran** [comment]
tested with sat 6.5
```
>>> a = entities.ForemanStatus().read()
2018-11-09 11:02:20 - nailgun.client - DEBUG - Making HTTP GET request to https://ibm-hs22-05.rhts.eng.brq.redhat.com/api/v2/status with options {'auth': ('admin', 'changeme'), 'verify': False, 'headers': {'content-type': 'application/json'}}, no params and no data.
2018-11-09 11:02:20 - nailgun.client - DEBUG - Received HTTP 200 response: {"result":"ok","status":200,"version":"1.20.0-RC2","api_version":2}
>>> a
nailgun.entities.ForemanStatus(result='ok', status=200, version='1.20.0-RC2', api_version=2)
>>> a = entities.KatelloStatus().read()
2018-11-09 11:02:36 - nailgun.client - DEBUG - Making HTTP GET request to https://ibm-hs22-05.rhts.eng.brq.redhat.com/katello/api/v2/status with options {'auth': ('admin', 'changeme'), 'verify': False, 'headers': {'content-type': 'application/json'}}, no params and no data.
2018-11-09 11:02:37 - nailgun.client - DEBUG - Received HTTP 200 response:   {"version":"3.9.0.rc2","timeUTC":"2018-11-09 09:02:37 UTC"}

>>> a
nailgun.entities.KatelloStatus(version='3.9.0.rc2', timeUTC='2018-11-09 09:02:37 UTC')
```

**Pull request merged**